packages/plugin-ext/src/plugin/quick-open.ts

```diff
@@ -613,9 +613,15 @@ export class QuickPickExt<T extends theia.QuickPickItem> extends QuickInputExt i
             this._handlesToItems.set(i, item);
             this._itemsToHandles.set(item, i);
         });
-        items.forEach((item, i) => Object.assign(item, { handle: i }));
         this.update({
-            items
+            items: items.map((item, i) => ({
```

Comment:
Why are you mapping exactly this set of properties? Is there a type definitions somewhere (or documentation) that tells us these are the right properties to map?

Reply:
These are all properties of the [`QuickPickItem`](https://github.com/eclipse-theia/theia/blob/master/packages/plugin/src/theia.d.ts#L2081-L2107) type. Added a missing property `type` in the last commit.